### ISSUE TITLE
Distinguish core and optional dependencies

### ISSUE THREAD

**Issue by TomJGooding** [davep/pispy#20]
It might be nice to distinguish between core and optional dependencies in the `Requires` section.

For example, the `textual` package shows tree-sitter and tree-sitter-languages which are optional extras. Other packages may have defined separate `dev` only dependencies.

**Comment by davep** [comment]
Aye, nice idea. I'll add that the next time I stream working on this; seems like a nice self-contained change.

**Comment by davep** [comment]
Note to self: see also #24 from @hugovk - like, sorting is a nice thing to do for folk! (Nobody ask why I didn't do that in the first place).

**Comment by davep** [comment]
Note to self: Also noting #25 too it might make sense to tidy up all of this by using [Packaging](https://packaging.pypa.io/en/stable/index.html) instead.

**Comment by davep** [comment]
Looking into the options, either I'm missing the obvious here, or neither of `parse_requirements` nor `Requirement` in Package provide a way of getting at that `extra` value:

```json
    "requires_dist": [
      "markdown-it-py[linkify,plugins]>=2.1.0",
      "rich>=13.3.3",
      "typing-extensions<5.0.0,>=4.4.0",
      "tree-sitter<0.21.0,>=0.20.1; extra == \"syntax\"",
      "tree-sitter-languages==1.10.2; extra == \"syntax\""
    ],
```

There also seems to be nothing in the metadata as acquired from the metadata that says "oh yeah there's this syntax thing you might want to say you want...".

**Comment by davep** [comment]
Yup, was playing around with that this morning. While I've not reached a final conclusion on on it, I think I'm falling down on the side of this not really being a handy thing for PISpy (albeit I am considering a consolidated vs detailed dependency display toggle).

**Comment by TomJGooding** [comment]
Sorry I hadn't really considered all the possible dependency "depends" when I suggested this! A consolidated vs detailed dependency display toggle would be a nice feature but only if it isn't too much work.
